File: 121_best_time_to_buy_and_sell_stock/121_best_time_to_buy_and_sell_stock.ts
function maxProfit(prices: number[]): number {
    let lowestPrice: number = Number.MAX_SAFE_INTEGER;
    let todayProfit: number = 0;
    let maxProfit: number = 0;
    
    for (let i: number = 0; i < prices.length; i++){
        if (lowestPrice > prices[i]){
            lowestPrice = prices[i];
        }
        
        todayProfit = prices[i] - lowestPrice;
        
        if (todayProfit > maxProfit){
            maxProfit = todayProfit;
        }
    }
    return maxProfit;
};

File: 121_best_time_to_buy_and_sell_stock/121_best_time_to_buy_and_sell_stock.go
package besttimetobuyandsellstock

import "math"

func maxProfit(prices []int) int {
	lowestPrice := math.MaxInt
	todayProfit := 0
	maxProfit := 0

	for i := 0; i < len(prices); i++ {
		if lowestPrice > prices[i] {
			lowestPrice = prices[i]
		}
		todayProfit = prices[i] - lowestPrice

		if todayProfit > maxProfit {
			maxProfit = todayProfit
		}
	}
	return maxProfit
}
